src/js/pyodide.ts

```diff
@@ -88,6 +88,21 @@ function finalizeBootstrap(API: any, config: ConfigType) {
 
   // Set up key Javascript modules.
   let importhook = API._pyodide._importhook;
+  importhook.add_all = (o: object) => {
```

Comment:
Maybe we can pass the `add_all` function as a parameter to `register_js_finder`? I think it will make it a bit clearer and extendable.

For example:

```py

class JsFinder(MetaPathFinder):
    def __init__(self) -> None:
        self.hook = lambda: pass
    def add_hook(self, hook):
        self.hook = hook
    def register_js_module():
        self.hook(jsproxy)

def register_js_finder(*, hook):
    ...
    jsfinder.add_hook(hook)
    ...
```

Reply:
Sounds good to me, thanks for the suggestion.